MixStream: g_mutex_free is deprecated

This warning was raised when compiling the MixStream extension:

```
fretwork/mixstream/MixStream.c: In function ‘mix_stream_destroy’:
fretwork/mixstream/MixStream.c:135:3: warning: ‘g_mutex_free’ is deprecated (declared at /usr/include/glib-2.0/glib/deprecated/gthread.h:275) [-Wdeprecated-declarations]
```

As we're using glib 2.34.3+, [`g_mutex_free` can be removed or replaced by `g_mutex_clear`][1].
Eventually, we can add a condition on the version of glib.

[1]: https://developer.gnome.org/glib/stable/glib-Deprecated-Thread-APIs.html#g-mutex-free

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 - fix: remove `g_thread_init` (deprecation) ([#26](https://github.com/fofix/python-mixstream/pull/26))
 - fix: replace `GStaticMutex` with `GMutex` (deprecation) ([#24](https://github.com/fofix/python-mixstream/pull/24))
+- fix: replace `g_mutex_free` with `g_mutex_clear` to free resources (deprecation) ([#29](https://github.com/fofix/python-mixstream/pull/29))
 - fix: replace `g_mutex_new` with `g_mutex_init` to init a mutex (deprecation) ([#28](https://github.com/fofix/python-mixstream/pull/28))
 
 ## 1.0.0 (2022-03-03)

--- a/mixstream/MixStream.c
+++ b/mixstream/MixStream.c
@@ -146,7 +146,7 @@ void mix_stream_destroy(MixStream* stream)
 {
   if (stream->channel != -1)
     mix_stream_stop(stream);
-  g_mutex_free(&stream->st_mutex);
+  g_mutex_clear(&stream->st_mutex);
   if (stream->soundtouch != NULL)
     soundtouch_delete(stream->soundtouch);
   if (stream->free_cb != NULL)

